backend/src/main/java/fk/prof/backend/request/profile/parser/MessageParser.java

```diff
@@ -1,16 +1,30 @@
 package fk.prof.backend.request.profile.parser;
 
+import com.codahale.metrics.Histogram;
 import com.google.common.io.ByteStreams;
 import com.google.protobuf.*;
 import fk.prof.backend.exception.AggregationFailure;
 import fk.prof.backend.request.CompositeByteBufInputStream;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
 
 import java.io.IOException;
 
 /**
  * @author gaurav.ashok
  */
-public abstract class MessageParser {
+public class MessageParser {
+
+    private static Logger logger = LoggerFactory.getLogger(MessageParser.class);
+    private Histogram histWseSize;
+
+    public MessageParser() {
```

Comment:
Do we need this constructor? Why not kill it and fold the null check into the constructor, so we can eliminate null-check in the impl?